src/components/questionnaire/questionnaire-interaction-screen.tsx

```diff
@@ -0,0 +1,160 @@
+import { KeyboardReturn } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  Typography
+} from "@mui/material";
+import { useState } from "react";
+import type { AnswerOption, Question, Questionnaire } from "src/generated/homeLambdasClient";
+import strings from "src/localization/strings";
+
+interface QuestionnaireInteractionScreenProps {
+  questionnaire: Questionnaire | null;
+  mode: "view" | "edit" | "preview";
+  onBack: () => void;
+}
+
+/**
+ *  Screen for the user to interact with the questionnaire; fill, edit and preview
+ */
+const QuestionnaireInteractionScreen: React.FC<QuestionnaireInteractionScreenProps> = ({
+  questionnaire,
+  mode,
+  onBack
+}) => {
+  if (!questionnaire) {
+    return <Typography variant="h4">Loading...</Typography>;
+  }
+
+  const [responses, setResponses] = useState<Record<string, string[]>>({});
+
+  const handleOptionChange = (questionId: string, optionLabel: string, isChecked: boolean) => {
+    setResponses((prevResponses) => {
+      const selectedOptions = prevResponses[questionId] || [];
+      return {
+        ...prevResponses,
+        [questionId]: isChecked
+          ? [...selectedOptions, optionLabel]
+          : selectedOptions.filter((label) => label !== optionLabel)
+      };
+    });
+  };
+
+  const handleSubmit = () => {
+    let correctAnswersCount = 0;
+    questionnaire.questions.forEach((question) => {
+      const userAnswers = responses[question.questionText] || [];
+      const correctAnswers = question.answerOptions
+        .filter((option) => option.isCorrect)
+        .map((option) => option.label);
+      if (
+        userAnswers.length === correctAnswers.length &&
```

Comment:
Does this condition imply that it is possible for a user to select more answers than the question has correct answers? If so we should prevent this behavior at the point of the user filling in the questionnaire- and then the first part of this condition can be removed.